cwltool/factory.py

```diff
@@ -68,3 +87,45 @@ def make(self, cwl: Union[str, dict[str, Any]]) -> Callable:
         if isinstance(load, int):
             raise WorkflowException("Error loading tool")
         return Callable(load, self)
+
+    def _fix_loading_context(self, runtime_context: RuntimeContext) -> None:
+        self.loading_context.resolver = getdefault(self.loading_context.resolver, tool_resolver)
+        self.loading_context.singularity = runtime_context.singularity
+        self.loading_context.podman = runtime_context.podman
+
+    def _fix_runtime_context(self) -> None:
+        self.runtime_context.basedir = os.getcwd()
+        self.runtime_context.find_default_container = functools.partial(
+            find_default_container, default_container=None, use_biocontainers=None
+        )
+
+        if sys.platform == "darwin":
+            default_mac_path = "/private/tmp/docker_tmp"
+            if self.runtimeContext.tmp_outdir_prefix == DEFAULT_TMP_PREFIX:
+                self.runtimeContext.tmp_outdir_prefix = default_mac_path
+
+        for dirprefix in ("tmpdir_prefix", "tmp_outdir_prefix", "cachedir"):
+            if (
+                getattr(self.runtime_context, dirprefix)
+                and getattr(self.runtime_context, dirprefix) != DEFAULT_TMP_PREFIX
+            ):
+                sl = (
+                    "/"
+                    if getattr(self.runtime_context, dirprefix).endswith("/")
+                    or dirprefix == "cachedir"
+                    else ""
+                )
+                setattr(
+                    self.runtime_context,
+                    dirprefix,
+                    os.path.abspath(getattr(self.runtime_context, dirprefix)) + sl,
+                )
+                if not os.path.exists(os.path.dirname(getattr(self.runtime_context, dirprefix))):
+                    try:
+                        os.makedirs(os.path.dirname(getattr(self.runtime_context, dirprefix)))
+                    except Exception as e:
+                        print("Failed to create directory: %s", e)
```

Comment:
Todo, refactor this and the equivalent code from `cwltool/main.py` to a shared function. May need to look back to https://github.com/common-workflow-language/cwltool/blob/b0db43e375815453902ee16cec096154a4ed185b/cwltool/main.py#L748 (when this PR was originally written) for inspiration